js/widgets.js

```diff
@@ -113,70 +178,149 @@ function handleIntegration (event) {
  * Main function retrieving the HTML code from the API.
  * Keep the chunk > 12 otherwise territories pages will issue more than one query.
  */
-function embedDatasets (territories, datasets) {
-  chunk(territories.concat(datasets), 12).forEach((elements) => {
+function embedDatasets (territories, datasets, dataTerritoryIdAttr, dataDatasetIdAttr) {
+  const items = territories.concat(datasets)
+  const chunkBy = 12
+  const promises = chunk(items, chunkBy).map((elements) => {
     const references = elements.map((el) => {
-      if (el.hasAttribute(TERRITORY_ID)) {
-        return `territory-${el.dataset.udataTerritoryId}`
-      } else if (el.hasAttribute(DATASET_ID)) {
-        return `dataset-${el.dataset.udataDatasetId}`
+      if (el.hasAttribute(dataTerritoryIdAttr)) {
+        return `territory-${el.dataset[camelCaseData(dataTerritoryIdAttr)]}`
+      } else if (el.hasAttribute(dataDatasetIdAttr)) {
+        return `dataset-${el.dataset[camelCaseData(dataDatasetIdAttr)]}`
       }
     })
     const url = `${baseURL}/api/1/oembeds/?references=${references}`
     // Warning: if you are tempted to use generators instead of chaining
     // promises, you'll have to use babel-polyfill which adds 300 Kb
     // once the file is converted to ES5.
-    fetch(url)
-      .then(checkStatus)
-      .then((response) => response.json())
+    return fetchJSON(url)
       .then((jsonResponse) => {
```

Comment:
I think you could have a better name than just `jsonResponse` here. Something like `datasetsContent` (not happy with this though). You are describing how you this variable got created (by a network request), it's usually better to describe what is in the variable.